core/src/stdlib/pyscript/util.py

```diff
@@ -31,3 +37,20 @@ def __setattr__(self, attr, value):
 
     def __call__(self, *args):
         raise TypeError(self.error)
+
+
+def is_awaitable(obj):
+    """
+    Returns a boolean indication if the passed in obj is an awaitable
+    function. (MicroPython treats awaitables as generator functions, and if
+    the object is a closure containing an async function we need to work
+    carefully.)
+    """
+    if "micropython" in sys.version.lower():
```

Comment:
see my previous comment ... we should really not use here and there the same check that is also not super reliable for custom runtimes out there (even if we might use the same trick internally, but that's why we should not propagate that elsewhere).

Reply:
Fixed in latest push.